### DISCUSSION TITLE
facing issue in type casting

### DISCUSSION THREAD

**Discussion by nouman073** [AammarTufail/python_ka_chilla2023#44] [Q&A]
The favourite_Dish column data type is "object"
I ran this code line df['favourite_Dish'].astype('str') but it does not change the datatype object to string
code running without error.
if run same on df['Height'].astype('int') it work properly

**Comment by Masoodullah909** [comment]
> The favourite_Dish column data type is "object" I ran this code line df['favourite_Dish'].astype('str') but it does not change the datatype object to string code running without error. if run same on df['Height'].astype('int') it work properly

The astype() method will only be able to convert the data type of a column if all of the values in the column can be safely converted to the desired data type. If any values in the column cannot be converted, the astype() method will return an error.


For Example: In string to Int Conversion
"5" can be converted into 5
but "Five" cannot be converted into 5, this will return an error


I hope you get my point.
